rustsat/src/instances/fio.rs

```diff
@@ -66,3 +75,167 @@ pub(crate) fn open_compressed_uncompressed_write<P: AsRef<Path>>(
+                };
+
+                current_assignment.extend_from_vline(&line)?;
+            } else {
+                solution = Some(Assignment::from_vline(&line)?);
+            }
+
+            have_vline = true;
+        }
+    }
+
+    //There is no solution line so we can not trust the output
+    if !is_sat {
+        return Err(anyhow::anyhow!(SatSolverOutputError::NoSline));
+    }
+
+    if have_vline {
+        if let Some(solution) = solution {
+            return Ok(SolverOutput::Sat(solution));
+        }
+    }
+
+    Err(anyhow::anyhow!(SatSolverOutputError::NoVline))
+}
+
+#[cfg(test)]
+mod tests {
+    use std::io;
+
+    use crate::{
+        instances::fio::SatSolverOutputError,
+        types::{Assignment, TernaryVal},
+    };
+
+    use super::{parse_sat_solver_output, SolverOutput};
+
+    #[test]
+    fn parse_solver_output_sat() {
+        let ground_truth = SolverOutput::Sat(Assignment::from(vec![
+            TernaryVal::True,
+            TernaryVal::False,
+            TernaryVal::DontCare,
+            TernaryVal::True,
+            TernaryVal::False,
+            TernaryVal::True,
+        ]));
+
+        let data = "c this is a comment\ns SATISFIABLE\nv 1 -2 4 -5 6 0\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader).unwrap();
+        assert_eq!(res, ground_truth);
+
+        let data = "c this is a comment\nv 1 -2 4 -5 6 0\ns SATISFIABLE\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader).unwrap();
+        assert_eq!(res, ground_truth);
+
+        let data = "c this is a comment\ns SATISFIABLE\nv 1 -2 4 \nv -5 6 0\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader).unwrap();
+        assert_eq!(res, ground_truth);
+    }
+
+    #[test]
+    fn parse_solver_output_unsat() {
+        let data = "c this is a comment\ns UNSATISFIABLE\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader).unwrap();
+        assert_eq!(res, SolverOutput::Unsat);
+    }
+
+    #[test]
+    fn parse_solver_output_unknown() {
+        let data = "c this is a comment\ns UNKNOWN\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader).unwrap();
+        assert_eq!(res, SolverOutput::Unknown);
+    }
+
+    #[test]
+    fn parse_solver_output_indeterminate() {
+        let data = "c this is a comment\ns INDETERMINATE\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader).unwrap();
+        assert_eq!(res, SolverOutput::Unknown);
+    }
+
+    #[test]
+    fn parse_solver_output_noslinewithvline() {
+        let data = "c this is a comment\nv 1 -2 4 -5 6 0\n";
+        let reader = io::Cursor::new(data);
+        let res = parse_sat_solver_output(reader);
+        assert!(res.is_err());
```

Comment:
```suggestion
        match res.downcast::<SolverOutputError>() {
            Some(err) => match err {
                SolverOutputError::NoSLine => (),
                _ => panic!()
            }
            Err(_) => panic!(),
        }
```
This (or something along those lines, I'm writing this free hand) is how you can check for the exact error type in the anyhow error. It would be great if you can do this for the other error cases as well.